Amethyst/Preferences/GeneralPreferencesViewController.swift

```diff
@@ -24,16 +24,18 @@ final class GeneralPreferencesViewController: NSViewController, NSTableViewDataS
     override func viewWillAppear() {
         super.viewWillAppear()
 
-        layouts = UserConfiguration.shared.layoutStrings()
+        layoutKeys = UserConfiguration.shared.layoutKeys()
 
         layoutsTableView?.reloadData()
     }
 
     @IBAction func addLayout(_ sender: NSButton) {
         let layoutMenu = NSMenu(title: "")
 
-        for layoutString in LayoutManager.availableLayoutStrings() {
-            let menuItem = NSMenuItem(title: layoutString, action: #selector(addLayoutString(_:)), keyEquivalent: "")
+        for (layoutKey, layoutName) in LayoutManager.availableLayoutStrings() {
+            let menuItem = NSMenuItem(title: layoutKey, action: #selector(addLayoutString(_:)), keyEquivalent: "")
+            menuItem.attributedTitle = NSAttributedString(string: layoutName)
+            menuItem.title = layoutKey
```

Comment:
I think this line is redundant with the initializer, but regardless is there a reason to keep the title at all since we're using the attributed title?

Reply:
The reason to keep the title is to hold the `layoutKey` somewhere on the `NSMenuItem`.  Specifically, the `action` (callback? not sure the Swift term) `addLayoutString()` needs to know the `layoutKey` for its own use, and has to get it from the `NSMenuItem`.  As you can see, `addLayoutString()` does use `sender.title` and not `sender.attributedTitle` (which is the name rather than the key).

Re-assigning the title does seem redundant with the initializer.  However, without this line, `addLayoutString()` sees `sender.title` as being the layout name rather than key, in my testing.  I hypothesize that perhaps setting the `attributedTitle` also overrides the `title`, so then we have to re-set the title.  I also note that pre-existing code already redundantly assigns the `menuItem.action` despite it being apparently set in the initializer, so I figured there might be something weird going on that I didn't understand with Swift initializers or with `NSMenuItem` in particular.

Reply:
You may be able to use this: https://developer.apple.com/documentation/appkit/nsmenuitem/1514834-representedobject

Reply:
Indeed, that's exactly what I needed.  Pushed a commit making this change and removing the `attributedTitle`.